### ISSUE TITLE
[BUDI-6597] Filter Not Applied to Data Provider

### ISSUE THREAD

**Issue by danderson2692** [Budibase/budibase#9736]
**Checklist**
- [X] I have searched budibase discussions and github issues to check if my issue already exists

**Hosting**
- Cloud
  - Tenant ID: dog.budibase.app

**Describe the bug**
Filter not applied to Data Provider using dynamic (state) or static values.

**To Reproduce**
1. I have a Data Provider as the top component on a details screen with the filter set to [ID] equals {{State.selectedDogID}}
2. Home screen in app has a Repeater with buttons that set the state to selectedDogID, then navigates to the Details screen
3. The Details screen always shows the first record in the table even when it's not the chosen record. This is also shown in the DevTools bindings

**Expected behavior**
The data provider should be filtered to the selected ID

**Screenshots**
Home screen button to set state and then navigate:
![SelectedID](https://user-images.githubusercontent.com/16950976/219902680-2a3eaf45-8279-486e-87f9-0930844b426f.PNG)

Filter applied to the Details data provider:
![image](https://user-images.githubusercontent.com/16950976/219902770-45b47f20-faca-4612-a754-642046117c6b.png)

Header showing the ID of the record and the State ID:
![image](https://user-images.githubusercontent.com/16950976/219902754-056661e1-eb47-42d4-8b7c-e1daa3318689.png)

Behavior of the app, showing different IDs:
![image](https://user-images.githubusercontent.com/16950976/219902814-3129844a-ba2f-4181-a704-d1e19d7d9325.png)

It performs the same way even if I hard-code the ID. I've also tried adding a button to manually refresh the data provider, and have tested this on a screen where the only components are the data provider and 2 text boxes.
![image](https://user-images.githubusercontent.com/16950976/219902829-3506584d-eed0-4ea3-93b3-ab6ab7252a65.png)
 
**App Export**
I can provide this privately if you'd like.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Browser: Chrome version 110.0.5481.104
 - Budibase version 2.3.16

<sub>[BUDI-6597](https://linear.app/budibase/issue/BUDI-6597/filter-not-applied-to-data-provider)</sub>

**Comment by melohagan** [comment]
Hey @danderson2692 

Could you send your app export to support@budibase.com please?

**Comment by melohagan** [comment]
Hey @danderson2692 

Thank you for the app export. I just noticed you are not persisting the app state. 

If you want app state to carry between screens or page refreshes, then you'll need to tick **Persist this value**.

<img width="1124" alt="Screenshot 2023-03-01 at 09 22 18" src="https://user-images.githubusercontent.com/101575380/222097092-84400681-5752-45c3-b190-6107464c6d0b.png">

Alternatively, you could pass the ID via the URL: https://docs.budibase.com/docs/url-parameters

It is worth noting however that URL variables can be encoded, and may need to be decoded on the other side.

**Comment by danderson2692** [comment]
Hi @melohagan

I've set the state to Persist but I'm still seeing the same result - just like when Persist was unchecked, I can see the state values on the second screen and they're displayed in the DevTools interface but the filter didn't apply.
![image](https://user-images.githubusercontent.com/16950976/222188757-e9c5dd78-46c4-4677-8111-c0fe17afce8b.png)

I've updated the basiclog_raw screen's data provider to pull from the "Dogs" table which has basically the same data but is a Budibase table rather than a google sheet, and I updated the filter to filter on its ID field. This worked as expected and did filter the Dogs table and shows the correct record. This is the case both with and without Persist enabled:
![image](https://user-images.githubusercontent.com/16950976/222190087-114a9f98-5b3e-40a7-9117-bdde6dece2ef.png)

I've tried "Starts With", "Like", and "Is In" to apply the filter with the google sheets data source without luck. The Dogs table was originally uploaded from a csv export of the same google sheet that's causing the problems now and the values definitely match without any stray characters. Any other suggestions?
